### PR TITLE
Fix #3

### LINKED ISSUE #3: Login and Cart Modal not working on Mobile
Issue for: https://spectrum.chat/moltin/help/buy-button-not-working-on-mobile~c0393354-54e9-47d4-be30-bbce7fc5565b

### PR DESCRIPTION
**Pull request by s0kil**

### CODE REVIEW

**Comment by notrab** [comment]
Hi @s0kil 

Nice idea!

While this would fix the issue for this specific use of `useOutsideClick` hook, it would only work in that instance and not for other items like we have in development for custom dropdowns.

**Comment by s0kil** [comment]
@notrab I have modified solution to be more general, should work for different use cases.

File: src/components/Modal/index.js [hunk 1]
```diff
@@ -1,17 +1,16 @@
...
-import { useStore, useActions } from 'easy-peasy'
-import { StripeProvider, Elements } from 'react-stripe-elements'
-import styled from 'styled-components'
-
-import useOnClickOutside from '../../hooks/useOnClickOutside'
-import useScript from '../../hooks/useScript'
+import { useActions, useStore } from 'easy-peasy'
 
-import Header from './Header'
+import Cart from '../Cart'
```
Comment (notrab) [inline src/components/Modal/index.js]: @s0kil is this a plugin that has changed the order of the imports?

Can we keep absolutes up top and then relative imports below (with hooks/lib/utils above components). I'll better document this in contributing notes.

File: src/hooks/useOnClickOutside.js [hunk 2]
```diff
@@ -1,21 +1,28 @@
...
 
-export default function useOnClickOutside(ref, handler) {
+export default function useOnClickOutside(ref, handler, open) {
+  const clickEvent = 'ontouchstart' in window ? 'touchstart' : 'mousedown'
+
   useEffect(() => {
     const listener = event => {
-      if (!ref.current || ref.current.contains(event.target)) {
+      if (ref.current.contains(event.target)) {
+        // Inside click, return
```
Comment (notrab) [inline src/hooks/useOnClickOutside.js]: Can we clear up this comment? Your code seems pretty clear to me what the outcome here is.

File: src/hooks/useOnClickOutside.js [hunk 3]
```diff
@@ -1,21 +1,28 @@
...
         return
       }
 
+      // Outside click, Handle event
```
Comment (notrab) [inline src/hooks/useOnClickOutside.js]: Can we clear up this comment? Your code seems pretty clear to me what the outcome here is.

File: src/hooks/useOnClickOutside.js [hunk 4]
```diff
@@ -1,21 +1,28 @@
...
       handler(event)
     }
 
-    document.addEventListener('mousedown', listener)
-    document.addEventListener('touchstart', listener)
+    // Optimize
```
Comment (notrab) [inline src/hooks/useOnClickOutside.js]: Can we clear up this comment? Your code seems pretty clear to me what the outcome here is.

**Review by notrab — CHANGES_REQUESTED** [review]
@s0kil I'm pretty happy with this now. Just a few things inside `Modal/index.js` that could be reverted to reduce the diff here.

Nice job and thanks for the PR 🚀

File: src/components/Modal/index.js (hunk 1 above)
Comment (s0kil) [reply src/components/Modal/index.js]: Yes, Plugin auto sorts imports by name.

**Review by notrab — APPROVED** [review]

**Comment by notrab** [comment]
@s0kil that's now merged & deployed 🚀 

Thank you for your contribution. Any further help would be massively appreciated.

[DM me on Twitter](https://twitter.com/notrab) and I'll send you some Moltin swag 😄

**Pull request merged**